src/__tests__/internal/Button.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { shallow, mount, render } from 'enzyme';
+import { expect } from 'chai';
+import sinon from 'sinon';
+import Button from '../../components/internal/Button';
+
+// Shallow Rendering
+// https://github.com/airbnb/enzyme/blob/master/docs/api/shallow.md
+describe('Button - Shallow Rendering', () => {
+  it('to have 1 .btn class', () => {
+    const wrapper = shallow(
+      <Button
+        id="loginButton"
+        size="wide"
+        text="Login"
+        onClick={() => alert('Clicked login button...')}
+        ariaLabel="Login Button"
+        type="submit"
+      />,
+    );
+    expect(wrapper.find('.btn')).to.have.length(1);
+  });
+
+  it('to be the correct size', () => {
+    const wrapper = shallow(
+      <Button
+        id="loginButton"
+        size="thin"
+        text="Login"
+        onClick={() => alert('Clicked login button...')}
+        ariaLabel="Login Button"
+        type="submit"
+      />,
+    );
+    expect(wrapper.find('.btn').hasClass('btn--thin')).to.equal(true);
+  });
+
+  it('to have the correct text', () => {
+    const wrapper = shallow(
+      <Button
+        id="loginButton"
+        size="wide"
+        text="Login"
+        onClick={() => alert('Clicked login button...')}
+        ariaLabel="Login Button"
+        type="submit"
+      />,
+    );
+    expect(wrapper.text()).to.equal('Login');
+  });
+
+  it('to show the loading spinner', () => {
+    const wrapper = shallow(
+      <Button
+        id="loginButton"
+        size="wide"
+        text="Login"
+        onClick={() => alert('Clicked login button...')}
+        ariaLabel="Login Button"
+        type="submit"
+        loading
+      />,
+    );
+    expect(wrapper.find('#spinner')).to.have.length(1);
+  });
```

Comment:
Add a test for a situation where a user clicks on the button whilst the loading spinner is present, this should not result in the `onClick` method being run again